### PR TITLE
Introducing error reporter.

### LINKED ISSUE #85: The extension should have better error logging
Especially due to the new changes where sourcemaps are used to open links back in the editor. Figuring out the correct mapping values to use is difficult.

We should use a similar mechanism to https://github.com/microsoft/vscode-edge-debug2 so that users can tell what url is being mapped to what file (or failing to map in some cases)

### PR DESCRIPTION
**Pull request by vidorteg**
This PR introduces an error reporting mechanism so that the user is aware of:
- feedback for the user from specific actions. e.g if a file is not mapped correctly, let the user know that it requires it's attention.
https://i.imgur.com/wbkEp88.png

- errors or bugs that are not intended behavior and were the user can take action e.g a file that could not be opened because it does not exist anymore
https://i.imgur.com/2DeqrnN.png

For error/bugs the user is prompted to file a bug in github.

Usage will be introduced in a following PR.
Fixes #85

### CODE REVIEW

File: src/errorReporter.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  static async showErrorDialog(
+    error: ErrorEventInterface): Promise<void> {
+
+    const template = `<!-- Please delete any private information -->
+        **Version:**
+        ${// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          vscode.extensions.getExtension('ms-edgedevtools.vscode-edge-devtools')?.packageJSON.version || 'N/A'}
+
+        **OS:**
+        <!-- Add your hosting platform e.g Microsoft Windows, WSL -->
```
Comment (bgoddar) [inline src/errorReporter.ts:26]: nit: I'd also mention MacOS here

File: src/errorReporter.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        **Additional context:**
+        <!-- Add any other context or about the problem or screenshots here. -->`;
+
+    // cannot do multiline due to:
+    // https://github.com/Microsoft/vscode/issues/48900
+    const answer = await vscode.window
+      .showErrorMessage(
+        `An error occurred:\n${error.title}\n${error.message}`,
+        ...['File a bug', 'Close dialog']
```
Comment (bgoddar) [inline src/errorReporter.ts:39]: Nit: I think Olivia brought this up as well, wdyt about removing the Close dialog, since there is already a big x button

**Review by bgoddar — COMMENTED** [review]
I have a couple of nits and a proposal:

I'm wondering if we should add telemetry reporting as part of the ErrorReporter class. A couple of places we report errors after launching an error code. It might make sense to do that as part of the ErrorReporter? The only think is then it would need to be aware of the telemetryReporter instance, so that might prove to be too complicated to make them combine, since this is currently a static class. I'm fine keeping them separate for now

**Comment by bgoddar** [comment]
> I have a couple of nits and a proposal:
> 
> I'm wondering if we should add telemetry reporting as part of the ErrorReporter class. A couple of places we report errors after launching an error code. It might make sense to do that as part of the ErrorReporter? The only think is then it would need to be aware of the telemetryReporter instance, so that might prove to be too complicated to make them combine, since this is currently a static class. I'm fine keeping them separate for now

I actually am leaning towards keeping telemetry out of this, so I'm going to update to approved with some nits

**Review by bgoddar — APPROVED** [review]
(updating review to approve with nits commented)

File: src/errorReporter.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        ${error.message}
+
+        **Additional context:**
+        <!-- Add any other context or about the problem or screenshots here. -->`;
+
+    // cannot do multiline due to:
+    // https://github.com/Microsoft/vscode/issues/48900
+    const answer = await vscode.window
+      .showErrorMessage(
+        `An error occurred:\n${error.title}\n${error.message}`,
```
Comment (bgoddar) [inline src/errorReporter.ts:38]: nit: should we even include the '\n' since it doesn't do anything?

Comment (vidorteg) [reply src/errorReporter.ts:38]: Thanks for the review @bgoddar , regarding the telemetry I made the same question myself at some point I even thought one was a superset of the other but eventually got to a similar conclusion; there a couple of high priority scenarios where is better to have them split, so most likely they just have an intersection, so it is better to keep them separate (at least ATM)

File: src/errorReporter.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        ${error.message}
+
+        **Additional context:**
+        <!-- Add any other context or about the problem or screenshots here. -->`;
+
+    // cannot do multiline due to:
+    // https://github.com/Microsoft/vscode/issues/48900
+    const answer = await vscode.window
+      .showErrorMessage(
+        `An error occurred: ${error.title}\n${error.message}`,
```
Comment (bgoddar) [inline src/errorReporter.ts:38]: nit: remove this \n as well

**Review by flynnolivia — APPROVED** [review]
This LGTM, +1 to the last nit Brandon mentioned

**Pull request merged**